### ISSUE TITLE
Service hung? icloud3 stopped updating 

### ISSUE THREAD

**Issue by marceloaf** [gcobb321/icloud3_v3#62]
I left homr today around 9:50AM and I was monitoring both v2.4.7 and v3b11. At around 10:01AM v3b11 throwed an error that my watch device went offline and then it came back online at 10:03AM. Apparently since the first event, the service didn't update watch's 
 position anymore. Meanwhile v2.4.7 was working and getting my watch positions as usual.

If you want/need I have the v3b11 debug logs

**v3b11 stopped getting my position near my destination zone. Picture below shows my destination zone with the dog icon. I was already there when I captured this pic.**
![v3 - 1035AM](https://user-images.githubusercontent.com/11551691/217282773-ccdcf660-ccf7-4f83-b596-a8417ddaf72d.jpg)

**While v2.4.7 had already registered my arrival on the destination zone.**
![v2 - 1036AM](https://user-images.githubusercontent.com/11551691/217283145-d4c428de-8e41-4308-83ea-2aa03a5b7c9f.jpg)

**At 11:17AM I was back home and v2.4.7 registered that:**
![v2 - 1117AM](https://user-images.githubusercontent.com/11551691/217283457-7e204bae-08a1-48ac-b5a4-522e9a649f20.jpg)

**10 minutes later at 11:27AM v3b11 was still showing me near my destination zone as if I hadn't arrived there** 
![v3 - 1127AM](https://user-images.githubusercontent.com/11551691/217284613-7013c62f-ee8d-46c3-b465-ebe789a0b067.jpg)

**Below the entire route by v2.4.7 (first) and v3b11 (second):**
![2023-02-07 (1)](https://user-images.githubusercontent.com/11551691/217284895-0e550d38-400b-4fe4-a5d4-a4fe41d4eee2.png)

![2023-02-07](https://user-images.githubusercontent.com/11551691/217285025-26fb3713-0b18-46f9-9d71-51d72e83484f.png)

**Comment by gcobb321** [comment]
Lillian's watch hasn't been used for a few days, is offline with no location data (gps=(0,0)). iC3 should be going into a error retry mode with a varying interval (15-secs, 1-min-, 5-min, etc). It did not do anything but just sat there. No retries and no tracking results. Working on it now and reworking the way offline and no location data works so it follows the same logic as an old location or poor gps does. That might be how v2.4.7 was operating. I'll know in a little while and let you know what I find out.

**Comment by marceloaf** [comment]
Great! Thanks!! I didn't do anything with my instance since my watch stopped reporting early this morning. During this time, I noticed other devices were reporting ok although my watch was still in the same position. Some minutes ago, I noticed that one of my sistser's phone also stopped reporting. Her iphone is using fmf  and my watch is family sharing. Not sure if this info is relevant... 

**She is in aecs zone since 1:06PM according to v2.4.7:**
![image](https://user-images.githubusercontent.com/11551691/217331872-9e2dc41f-cb49-4987-a5fa-cfab74a580f1.png)

**The last time v3b11 reported it was 12:59PM really close to enter in aecs zone:**
![image](https://user-images.githubusercontent.com/11551691/217332204-977c96a2-35e4-4136-b331-d7137057219b.png)
